### PR TITLE
drivers: fpga: Place API into iterable section

### PR DESCRIPTION
**Pull request by pdgendt**
Add wrapper DEVICE_API macro to all fpga_driver_api instances.

### CODE REVIEW

**Review by cfriedt — COMMENTED** [review]
Looks good, but I would suggest holding off until @benediktibk 's ice40 PR is merged and then rebasing to avoid churn.

**Comment by pdgendt** [comment]
> Looks good, but I would suggest holding off until @benediktibk 's ice40 PR is merged and then rebasing to avoid churn.

Sure, also note that the `DEVICE_API` macro is already in `main`, new drivers can already make use of it.

File: drivers/flash/flash_sam.c [hunk 1]
```diff
@@ -464,7 +464,7 @@ static void sam_flash_api_pages_layout(const struct device *dev,
 	*layout_size = config->pages_layouts_size;
 }
 
-static struct flash_driver_api sam_flash_api = {
+static DEVICE_API(flash, sam_flash_api) = {
```
Comment (kartben) [inline drivers/flash/flash_sam.c:467]: Doesn't belong to this commit :)

**Review by kartben — CHANGES_REQUESTED** [review]

File: drivers/flash/flash_sam.c (hunk 1 above)
Comment (pdgendt) [reply drivers/flash/flash_sam.c:467]: Whoops

**Comment by pdgendt** [comment]
Removed flash driver, rebased and added ice40 fpga drivers.

**Review by kartben — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**